src/solvers/flattening/boolbv.cpp

```diff
@@ -580,23 +580,11 @@ bool boolbvt::boolbv_set_equality_to_true(const equal_exprt &expr)
 
 void boolbvt::set_to(const exprt &expr, bool value)
 {
-  if(expr.type().id()!=ID_bool)
-  {
-    error() << "boolbvt::set_to got non-boolean operand: "
-            << expr.pretty() << eom;
-    throw 0;
-  }
-
-  if(value)
-  {
-    if(expr.id()==ID_equal)
-    {
-      if(!boolbv_set_equality_to_true(to_equal_expr(expr)))
-        return;
-    }
-  }
-
-  return supert::set_to(expr, value);
+  PRECONDITION(expr.type().id() == ID_bool);
+  if(
+    !value || expr.id() != ID_equal ||
```

Comment:
This seems a lot more convoluted.

Reply:
Agreed that's a bit complicated. I added a fixup commit to have fewer negations.